driver-sync/src/test/functional/com/mongodb/client/unified/TestLoggingInterceptor.java

```diff
@@ -49,4 +55,20 @@ public void close() {
     public synchronized List<StructuredLogMessage> getMessages() {
         return new ArrayList<>(messages);
     }
+
+    public static class LoggingFilter{
+        private final Map<StructuredLogMessage.Component, StructuredLogMessage.Level> componentLevelMap;
+
+        public LoggingFilter(){
+            componentLevelMap = new HashMap<>();
+        }
+        public void addComponent(final StructuredLogMessage.Component component, final StructuredLogMessage.Level level){
+            componentLevelMap.put(component, level);
+        }
+
+        public boolean match(final StructuredLogMessage message){
+            StructuredLogMessage.Level level = componentLevelMap.get(message.getComponent());
+            return level == message.getLevel();
+        }
```

Comment:
Good catch, thanks! While I appreciate your suggestion to use compareTo(), I am hesitant to rely on the ordinal values as the basis for comparisons.

if we refer to the javadoc for ordinal method in Enum.java:
> Most programmers will have no use for this method. It is designed for use by sophisticated enum-based data structures, such as java.util.EnumSet and java.util.EnumMap.

What i would suggest using is a custom field in the enum to store the severity level, which is more or less general practice to avoid any potential issues in future. 

 `    
 public enum Level {
        DEBUG(1);

        private int severity;

        Level(int severity) {
            this.severity = severity;
        }

        public int getSeverity() {
            return severity;
        }
    }`
    
Let me know your thoughts on this approach

Reply:
Having our own field may indeed be better. As an alternative to exposing that field via a getter, we may instead expose a comparator via a `static` method (this approach is more like OOP: instead of exposing internals, we expose only the behavior we need). Something like

```java
public static Comparator<Level> severityComparator() {
    return COMPARATOR;
}
```

Reply:
Looking at the `Enum` javadoc, I don't see a reason not to just use the `Enum#compareTo` method.  The text that @vbabanin referred to above is from the `ordinal` method's javadoc.  There is no such warning about the `compareTo` method.  If programmers weren't meant to compare enum values, then presumably `Enum` would not implement `Comparable`.

Reply:
As discussed, we will proceed with using `compareTo`